### ISSUE TITLE
[CI] DynamicMappingIT testDynamicRuntimeObjectFields failing

### ISSUE THREAD

**Issue by fcofdez** [elastic/elasticsearch#81440]
I think this was fixed in https://github.com/elastic/elasticsearch/pull/80999 but it wasn't backported to 7.15

**Build scan:**
https://gradle-enterprise.elastic.co/s/7rzue7sl666co/tests/:server:internalClusterTest/org.elasticsearch.index.mapper.DynamicMappingIT/testDynamicRuntimeObjectFields

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.index.mapper.DynamicMappingIT.testDynamicRuntimeObjectFields" -Dtests.seed=F7CF20830793D357 -Dtests.locale=es-UY -Dtests.timezone=Mexico/BajaSur -Druntime.java=8`

**Applicable branches:**
7.15

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.index.mapper.DynamicMappingIT&tests.test=testDynamicRuntimeObjectFields

**Failure excerpt:**
```
junit.framework.AssertionFailedError: Unexpected exception type, expected MapperParsingException but got java.lang.IllegalArgumentException: mapper [obj.runtime.dynamic.number] cannot be changed from type [long] to [text]

  at __randomizedtesting.SeedInfo.seed([F7CF20830793D357:21079CF641ECBBFF]:0)
  at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2746)
  at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2734)
  at org.elasticsearch.index.mapper.DynamicMappingIT.testDynamicRuntimeObjectFields(DynamicMappingIT.java:432)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

  Caused by: java.lang.IllegalArgumentException: mapper [obj.runtime.dynamic.number] cannot be changed from type [long] to [text]

    at org.elasticsearch.index.mapper.FieldMapper.checkIncomingMergeType(FieldMapper.java:359)
    at org.elasticsearch.index.mapper.FieldMapper.merge(FieldMapper.java:344)
    at org.elasticsearch.index.mapper.FieldMapper.merge(FieldMapper.java:49)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:376)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:337)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:363)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:337)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:363)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:337)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:363)
    at org.elasticsearch.index.mapper.RootObjectMapper.doMerge(RootObjectMapper.java:314)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:337)
    at org.elasticsearch.index.mapper.RootObjectMapper.merge(RootObjectMapper.java:309)
    at org.elasticsearch.index.mapper.Mapping.merge(Mapping.java:138)
    at org.elasticsearch.index.mapper.MapperService.mergeMappings(MapperService.java:488)
    at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.applyRequest(MetadataMappingService.java:125)
    at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.execute(MetadataMappingService.java:82)
    at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:706)
    at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:328)
    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:223)
    at org.elasticsearch.cluster.service.MasterService.access$000(MasterService.java:63)
    at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:155)
    at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
    at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:678)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:259)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:222)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)

```

**Comment by cbuescher** [comment]
Yes, looks like it. I will take a look and maybe backport the other fix if needed.

**Comment by cbuescher** [comment]
Backport is open at https://github.com/elastic/elasticsearch/pull/81459

**Comment by cbuescher** [comment]
Should be fixed now by  28d4f40
